code/__DEFINES/cooldowns.dm

```diff
@@ -0,0 +1,25 @@
+#define COOLDOWN_CHEW 		"chew"
+#define COOLDOWN_PUKE 		"puke"
+#define COOLDOWN_POINT 		"point"
+#define COOLDOWN_EMOTE		"emote"
+#define COOLDOWN_VENTCRAWL	"ventcrawl"
+#define COOLDOWN_BUCKLE		"buckle"
+#define COOLDOWN_RESIST		"resist"
+#define COOLDOWN_ORDER		"order"
+#define COOLDOWN_DISPOSAL	"disposal"
+#define COOLDOWN_ACID		"acid"
+#define COOLDOWN_GUT		"gut"
+#define COOLDOWN_ZOOM		"zoom"
+#define COOLDOWN_BUMP		"bump"
+#define COOLDOWN_ENTANGLE	"entangle"
+#define COOLDOWN_NEST		"nest"
+#define COOLDOWN_TASTE		"taste"
+#define COOLDOWN_VENTSOUND	"vendsound"
+
+#define COOLDOWN_START(cd_source, cd_index, cd_time) LAZYSET(cd_source.cooldowns, cd_index, addtimer(CALLBACK(src, /atom/movable.proc/end_cooldown, cd_index), cd_time))
+
+#define COOLDOWN_CHECK(cd_source, cd_index) (cd_source.cooldowns && cd_source.cooldowns[cd_index])

```

Comment:
```suggestion
#define COOLDOWN_CHECK(cd_source, cd_index) LAZYACCESS(cd_source, cd_index)
```

Reply:
I had not done this initially because `LAZYACCESS` adds a few irrelevant checks, but thinking again that's sort of micro-optimizing and this is more clear and maintainable